addons/l10n_cl/models/res_partner.py

```diff
@@ -48,12 +48,13 @@ def create(self, values):
         return super().create(values)
 
     def write(self, values):
-        for record in self:
-            vat_values = {
-                'vat': values.get('vat', record.vat),
-                'l10n_latam_identification_type_id': values.get(
-                    'l10n_latam_identification_type_id', record.l10n_latam_identification_type_id.id),
-                'country_id': values.get('country_id', record.country_id.id)
-            }
-            values['vat'] = self._format_vat_cl(vat_values)
+        if any(field in values for field in ['vat', 'l10n_latam_identification_type_id', 'country_id']):
+            for record in self:
+                vat_values = {
+                    'vat': values.get('vat', record.vat),
+                    'l10n_latam_identification_type_id': values.get(
+                        'l10n_latam_identification_type_id', record.l10n_latam_identification_type_id.id),
+                    'country_id': values.get('country_id', record.country_id.id)
+                }
+                values['vat'] = self._format_vat_cl(vat_values)
```

Comment:
Changing the same value in a loop? :sus:

`for record in self`, and then use of `self`? :sus: